aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -878,6 +879,87 @@ impl AptosVM {
             traversal_context,
         )
     }
+    fn execute_automation_transaction<'a, 'r, 'l>(
+        &'l self,
+        resolver: &'r impl AptosMoveResolver,
+        mut session: UserSession<'r, 'l>,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext<'a>,
+        txn_data: &TransactionMetadata,
+        payload: &'a AutomationTransactionPayload,
+        log_context: &AdapterLogSchema,
+        new_published_modules_loaded: &mut bool,
+        change_set_configs: &ChangeSetConfigs,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        fail_point!("aptos_vm::execute_automation_transaction", |_| {
+            Err(VMStatus::Error {
+                status_code: StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                sub_status: Some(move_core_types::vm_status::sub_status::unknown_invariant_violation::EPARANOID_FAILURE),
+                message: None,
+            })
+        });
+        if !payload.is_valid() {
+            return Err(VMStatus::error(
+                StatusCode::INVALID_AUTOMATION_PAYLOAD,
+                Some(AutomationTransactionPayload::entry_function_reference()),
+            ));
+        }
+
+        gas_meter.charge_intrinsic_gas_for_transaction(txn_data.transaction_size())?;
+        if txn_data.is_keyless() {
+            gas_meter.charge_keyless()?;
+        }
```

Comment:
Has this been copied from some existing function? As of now we don't support keyless logins.

Reply:
yes, I kept the same flow as for SignedTranansaction which is not considered non-multisig. So if we do not support it now, I just kept it for future.